playlist_update.py

```diff
@@ -209,25 +213,35 @@ def refresh_the_token(client_id, client_secret):
             print(pgrm_signature + "the time left on new token is: "+ str(new_expire / 60) + "min") #says how long
             return response_json
         else:
+            config_tools.logs("ERROR! when attempting to run refresh_the_token()! The response we got was: "+ str(response))
             print(pgrm_signature + "ERROR! The response we got was: "+ str(response))
-            return TOKEN #last ditch to try and make it work if TOKEN varaible is still active
+            return
+            # return TOKEN #last ditch to try and make it work if TOKEN varaible is still active
 
 
 def refresh_sp(init_spotify_flag): 
     #get default config data
-    config_data = config_tools.config_data()
+    config_data = config_tools.config_data('setup.json')
     client_id = (config_data['client_id'])
     client_secret = (config_data['client_secret'])
     
+
     #gets spotify.json data
-    with open("spotify.json", 'r') as info: #reads the json file that was just written to 
-        data = json.load(info)
-        expires_in = (data['expires_in'])
-        expires_at = (data['expires_at'])
-        TOKEN = (data['access_token'])
-    
+    if(str(init_spotify_flag).upper() == "FALSE"):
+        with open("spotify.json", 'r') as info: #reads the json file that was just written to 
+            data = json.load(info)
+            expires_in = (data['expires_in'])
+            expires_at = (data['expires_at'])
+            TOKEN = (data['access_token'])
+    else:
+        #if the init spotify flag has not be set to true yet, have the .cache file write to spotify.json
+        cache_2_json(init_spotify_token_flag=init_spotify_flag) #call this in the spotbot.py file main need to improve token
```

Comment:
To elaborate on this; this should be done on initialization to populate the Spotify.json file on boot every time ensuring Spotify.json has info every time. 

Or it can be kept here and just check if this is the first time, it would be slightly more efficient to do on boot tho @karltrowbridge1